pkg/controller/servicebindingrequest/servicebinder.go

```diff
@@ -258,10 +277,55 @@ func (b *ServiceBinder) Bind() (reconcile.Result, error) {
 	}
 	sbrStatus.Secret = secretObj.GetName()
 
+	if isApplicationSelectorEmpty(b.SBR.Spec.ApplicationSelector) {
+		conditionsv1.SetStatusCondition(&sbrStatus.Conditions, conditionsv1.Condition{
+			Type:   CollectionReady,
+			Status: corev1.ConditionTrue,
+		})
+		conditionsv1.SetStatusCondition(&sbrStatus.Conditions, conditionsv1.Condition{
+			Type:    InjectionReady,
+			Status:  corev1.ConditionFalse,
+			Reason:  EmptyApplicationSelectorReason,
+			Message: ErrEmptyApplicationSelector.Error(),
+		})
+
+		// updating status of request instance
+		sbr, err := b.updateStatusServiceBindingRequest(b.SBR, sbrStatus)
+		if err != nil {
+			return RequeueOnConflict(err)
+		}
+
+		// appending finalizer, should be later removed upon resource deletion
+		sbr.SetFinalizers(append(removeStringSlice(b.SBR.GetFinalizers(), Finalizer), Finalizer))
+		if _, err = b.updateServiceBindingRequest(sbr); err != nil {
+			return NoRequeue(err)
+		}
+
+		b.Logger.Info(ErrEmptyApplicationSelector.Error())
+		return Done()
+	}
 	updatedObjects, err := b.Binder.Bind()
 	if err != nil {
 		b.Logger.Error(err, "On binding application.")
-		return b.onError(err, b.SBR, sbrStatus, updatedObjects)
+		if errors.Is(err, ErrApplicationNotFound) {
+			conditionsv1.SetStatusCondition(&sbrStatus.Conditions, conditionsv1.Condition{
+				Type:   CollectionReady,
+				Status: corev1.ConditionTrue,
+			})
+			conditionsv1.SetStatusCondition(&sbrStatus.Conditions, conditionsv1.Condition{
+				Type:    InjectionReady,
+				Status:  corev1.ConditionFalse,
+				Reason:  ApplicationNotFoundReason,
+				Message: ErrApplicationNotFound.Error(),
+			})
+
+			// appending finalizer, should be later removed upon resource deletion
+			b.SBR.SetFinalizers(append(removeStringSlice(b.SBR.GetFinalizers(), Finalizer), Finalizer))
+			if _, updateErr := b.updateServiceBindingRequest(b.SBR); updateErr != nil {
+				return NoRequeue(updateErr)
+			}
+		}
+		return b.onError(err, b.SBR, sbrStatus, nil)
```

Comment:
On similar note as another comment, this block might deserve its own function (sharing behavior with the `handleApplicationSelectorEmpty` method).

Reply:
Done 1b275f9